### PR TITLE
Launchpad Search: Searches only when nothing is found

### PR DESCRIPTION
**Pull request by sergeibbb**
(#3763)

# Checklist

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [x] My changes include any required corresponding changes to the documentation (including CHANGELOG.md and README.md)
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

### CODE REVIEW

File: src/plus/launchpad/launchpad.ts [hunk 1]
```diff
@@ -630,6 +614,15 @@ export class LaunchpadCommand extends QuickCommand<State> {
...
 
+				// wait a little bit because the active quickpick is updated with some delay
+				await new Promise(resolve => setTimeout(resolve, 100));
+				const hasActiveLaunchpadItems = quickpick.activeItems.find(i => 'item' in i);
+				if (hasActiveLaunchpadItems) {
+					// We have an active item, so we can exit now without sending any requests to API
+					this.updateItemsDebouncer.cancel();
+					return true;
+				}
+
```
Comment (sergiolms) [inline src/plus/launchpad/launchpad.ts:625]: Could this be handled by the [`onDidChangeActive`](https://code.visualstudio.com/api/references/vscode-api#QuickPick:~:text=onDidChangeActive%3A%20Event%3Creadonly%20T%5B%5D%3E) event?

```typescript
onDidChangeActive: async quickpick => {
	// Not sure if this needs an await
	const hasActiveLaunchpadItems = await quickpick.activeItems.find(i => 'item' in i);
	if (hasActiveLaunchpadItems) {
		this.updateItemsDebouncer.cancel();
		return true;
	}
},
```

Comment (sergeibbb) [reply src/plus/launchpad/launchpad.ts:625]: @sergiolms 
There are following problems:
1. I'm not sure that it's guaranteed that `onDidChangeActive` is called after than `onDidChangeValue`.
2. If after typing, then `onDidChangeValue` won't probably be called at all.
3. It just never gets executed:

**Comment by sergiolms** [comment]
Hey @sergeibbb, while reviewing your PR, I'd suggest the following code changes:

👉 [**Code Suggestion for #3770**](https://staging.gitkraken.dev/drafts/37682962-826b-4c04-b570-596ca24e2d1e?source=PRComment)
>https://github.com/gitkraken/vscode-gitlens/pull/3770

You can also review and apply these suggestions [locally on your machine](https://staging.gitkraken.dev/link/drafts/37682962-826b-4c04-b570-596ca24e2d1e?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPQS1NUnVNNkNQb2ZpIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://staging.gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://staging.gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

File: src/plus/launchpad/launchpad.ts (hunk 1 above)
Comment (sergiolms) [reply src/plus/launchpad/launchpad.ts:625]: Please take a look at the code suggestion here https://github.com/gitkraken/vscode-gitlens/pull/3770#issuecomment-2485910711 and test if it works for you too. This is working for me:

https://github.com/user-attachments/assets/10479abc-f4e1-401b-8c8c-6717cb0d81f6

**Review by sergiolms — APPROVED** [review]
It's working for me, I couldn't reproduce the reported issue anymore

**Pull request closed without merging**